conans/model/build_info.py

```diff
@@ -290,12 +335,35 @@ def __init__(self, ref_name, root_folder, default_values=None):
             self.builddirs.append(self._default_values.builddir)
         if self._default_values.frameworkdir is not None:
             self.frameworkdirs.append(self._default_values.frameworkdir)
-        self.components = DefaultOrderedDict(lambda: Component(self.rootpath,
-                                                               self.version, self._default_values))
+
+        self.components = DefaultOrderedDict(lambda: Component(self._rootpath,
```

Comment:
The constructor now receives the filter_empty, so any changes in the variable are propagated when we create a new component in the `package_info`